test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/MemberExpressionCacheKeyTest.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Reflection;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    public class MemberExpressionCacheKeyTest
+    {
+        [Fact]
+        public void GetEnumerator_ReturnsMembers()
+        {
+            // Arrange
+            var expected = new[] 
+            {
+                typeof(TestModel3).GetProperty(nameof(TestModel3.Value)),
+                typeof(TestModel2).GetProperty(nameof(TestModel2.TestModel3)),
+                typeof(TestModel).GetProperty(nameof(TestModel.TestModel2)),
+            };
+
+            var key = GetKey(m => m.TestModel2.TestModel3.Value);
+
+            // Act
+            var actual = GetMembers(key);
+
+            // Assert
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void GetEnumerator_WithNullableType_ReturnsMembers()
```

Comment:
Not clear how `object` versus `struct` versus `Nullable<T>` matters to `MemberExpressionCacheKey`. Why do these variations exist?

Reply:
Is having additional test variants an actual concern now?